### ISSUE TITLE
[introspection][dotnet] Enable LegacyAttributes test

### ISSUE THREAD

**Issue by spouliot** [xamarin/xamarin-macios#11055]
A new test is being added, but disabled, inside PR#10580. It detects the use of the old, legacy, availability attributes. Those should not be used inside the platform assemblies shipping for dotnet.

However this requires large changes in most manual [1] binding files and is better done in several, distinct PRs.

Once the conversion is complete then the test needs to be enabled to avoid regressions.

https://github.com/xamarin/xamarin-macios/pull/10580/files#diff-0f2fc77f14b962b0c7827112e11d4af3917f549422a5504f311356a0daeaf301R394

```
#if IOS || TVOS
	[Ignore ("work in progress")]
#endif
```

The test-side conversion also needs to be removed from `tests/common/PlatformInfo.cs`

https://github.com/xamarin/xamarin-macios/pull/10580/files#diff-3070012c3058ef2b1ff75465e1394c3c3a03eddeaddc34e8ac35c92e2920383eR147

```sharp
// FIXME - temporary, while older attributes co-exists (in manual bindings)
if (ca is AvailabilityBaseAttribute old)
	list.Add (old);
```

[1] the generator handles the conversion to the new dotnet attributes

**Comment by spouliot** [comment]
Update

MacCatalyst: LegacyAttributes: 1111 API with mixed legacy availability attributes
tvOS: LegacyAttributes : 942 API with mixed legacy availability attributes 
iOS: LegacyAttributes : 1236 API with mixed legacy availability attributes 

The intersection between them would be quite high.

Mac introspection for dotnet is not yet part of xharness

**Comment by rolfbjarne** [comment]
> A new test is being added, but disabled, inside PR#10580. It detects the use of the old, legacy, availability attributes. Those should not be used inside the platform assemblies shipping for dotnet.

I wonder if we could rewrite the old availability attributes with a C# source generator?

**Comment by spouliot** [comment]
Maybe ? 🤔 I had a look a while ago, IIRC...

Generating the new attributes was not the problem - at least with C#9, where most restrictions were removed on partial methods, but still missing on a few things

> The partial modifier is not available on delegate or enumeration declarations.
https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/partial-classes-and-methods

Removing the old/existing attributes is... but the linker can remove them. However it makes testing missing attributes more difficult for introspection.

**Comment by rolfbjarne** [comment]
This was fixed here: e16d8fe1fe3f723d4ebfd4fde56441fc4e4f3894
